### ISSUE TITLE
memory leak audit

### ISSUE THREAD

**Issue by bookshelfdave** [basho-labs/riak-c-client#55]
### Messages:
- [x] riak_get
- [x] riak_delete
- [ ] riak_ping
- [x] riak_put  
- [x] riak_2index  
- [x] riak_get_bucketprops
- [x] riak_mapreduce
- [x] riak_search  
- [x] riak_get_clientid
- [x] riak_serverinfo
- [ ] riak_error  
- [x] riak_listbuckets  
- [x] riak_set_bucketprops
- [x] riak_listkeys  
- [ ] riak_reset_bucketprops
- [x] riak_set_clientid
### General:
- [x] riak_object
- [x] riak_binary
- [ ] riak_connection
- [ ] riak_log_config
- [ ] riak_types
- [ ] riak  
- [x] riak_bucketprops
- [ ] riak_error  
- [ ] riak_messages
- [ ] riak_operation
- [ ] riak_async  
- [ ] riak_config
- [ ] riak_log  
- [ ] riak_network

**Comment by hazen** [comment]
Is this a hand job or did you have an automated tool in mind?

**Comment by bookshelfdave** [comment]
:fish:

**Comment by bookshelfdave** [comment]
I have a couple ideas, maybe we can discuss tomorrow. I liked @bkerley's idea to check Valgrind output automatically. As a first pass, we should check for riak_free's vs riak_binary_free's.

**Comment by hazen** [comment]
Yes, I also have a couple of ideas.  We should figure out what we want to run Valgrind on, too.

**Comment by bookshelfdave** [comment]
See: https://github.com/basho/riak-c-client/pull/59

**Comment by bookshelfdave** [comment]
@javajolt I don't see where operation specific requests are being freed outside of [riak_operation_free ](https://github.com/basho/riak-c-client/blob/develop/src/riak_operation.c#L50).

For example, what takes care of freeing this binary?
https://github.com/basho/riak-c-client/blob/develop/src/internal/messages/riak_set_clientid-internal.h#L26

**Comment by bookshelfdave** [comment]
Messages with custom requests:
- set client id
- set bucketprops
- get bucketprops
- reset bucketprops
